sklearn/linear_model/tests/test_bayes.py

```diff
@@ -181,9 +183,9 @@ def test_update_of_sigma_in_ard():
     y = np.array([0, 0])
     clf = ARDRegression(n_iter=1)
     clf.fit(X, y)
-    # With the inputs above, ARDRegression prunes one of the two coefficients
-    # in the first iteration. Hence, the expected shape of `sigma_` is (1, 1).
-    assert clf.sigma_.shape == (1, 1)
+    # With the inputs above, ARDRegression prunes both of the two coefficients
+    # in the first iteration. Hence, the expected shape of `sigma_` is (0, 0).
+    assert clf.sigma_.shape == (0, 0)
```

Comment:
If the target is constant, it makes sense to me that both coefficients should bet set to zero. So I think this is a bug fix

Reply:
By setting `clf.sigma_ = np.array([])`, the later call to `clf.predict(X, return_std=True)` would return a different result for the `std` when compared to master. Is this okay?

Reply:
yes, it's a bug fix. Sigma should be nothing when keep_lambda is all false.

You might want to slightly delay your review @thomasjpfan , I'm going to update this for the n_features > n_samples case